forecasting/src/autogluon/forecasting/utils/dataset_utils.py

```diff
@@ -0,0 +1,297 @@
+import logging
+from copy import deepcopy
+
+import pandas as pd
+from gluonts.dataset.repository.datasets import dataset_recipes
+from gluonts.dataset.common import ListDataset
+from gluonts.dataset.field_names import FieldName
+
+from autogluon.common.features.types import R_INT, R_FLOAT, R_CATEGORY, R_OBJECT
+from autogluon.features.generators import IdentityFeatureGenerator, CategoryFeatureGenerator
+
+__all__ = [
+    'TimeSeriesDataset',
+    'gluonts_builtin_datasets',
+    'rebuild_tabular',
+    'time_series_dataset',
+    'train_test_split_dataframe',
+    'train_test_split_gluonts'
+]
+
+logger = logging.getLogger()
+
+
+class TimeSeriesDataset(ListDataset):
```

Comment:
There are no tests for this important class at all, I would add a big todo here.